From 54e5751a1810be7cec7ad8e63d7fa421d74a898b Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:09 -0400
Subject: [virt] kvm/x86: Break kvm_for_each_vcpu loop after finding the VP_INDEX

Message-id: <1406040016-3289-155-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86233
O-Subject: [RHEL7 PATCH v2 154/221] KVM: x86: Break kvm_for_each_vcpu loop after finding the VP_INDEX
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Takuya Yoshikawa <yoshikawa_takuya_b1@lab.ntt.co.jp>

No need to scan the entire VCPU array.

Signed-off-by: Takuya Yoshikawa <yoshikawa_takuya_b1@lab.ntt.co.jp>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 684851a15744355f294ee3fee4ca2e9108382b47)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 46dd031..8d0754e 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -2361,9 +2361,12 @@ static int get_msr_hyperv(struct kvm_vcpu *vcpu, u32 msr, u64 *pdata)
  case HV_X64_MSR_VP_INDEX: {
   int r;
   struct kvm_vcpu *v;
-  kvm_for_each_vcpu(r, v, vcpu->kvm)
-   if (v == vcpu)
+  kvm_for_each_vcpu(r, v, vcpu->kvm) {
+   if (v == vcpu) {
     data = r;
+    break;
+   }
+  }
   break;
  }
  case HV_X64_MSR_EOI:
-- 
1.7.1